internal/dcache/cluster_manager/cluster_manager_impl.go

```diff
@@ -266,6 +281,39 @@ func evaluateReadOnlyState() bool {
 	return false
 }
 
+func (cmi *ClusterManagerImpl) punchHeartBeat(rvList []dcache.RawVolume) {
+	hostname, err := os.Hostname()
+	common.Assert(err != nil, "Error getting hostname in punchHeartBeat  %v", err)
+	listMyRVs(rvList)
+	hbData := dcache.HeartbeatData{
+		IPAddr:        rvList[0].IPAddress,
+		NodeID:        cmi.nodeId,
+		Hostname:      hostname,
+		LastHeartbeat: uint64(time.Now().Unix()),
+		RVList:        rvList,
+	}
+
+	// Marshal the data into JSON
+	data, err := json.MarshalIndent(hbData, "", "  ")
+	//Adding Assert because error capturing can just log the error and continue because it's a schedule method
+	common.Assert(err != nil, "Error marshalling heartbeat data %+v : error - %v", hbData, err)
```

Comment:
pls use Sprintf to pass a formatted string to Assert.
also, since MarshalIndent is an external function, though it should not fail, but if it fails for whatever reason, we should handle it. In this case, we should not send the heartbeat, and log an error.